addons/test_mail/tests/test_mail_mail.py

```diff
@@ -823,6 +830,97 @@ def test_mail_mail_values_unicode(self):
         self.assertEqual(self._mails[0]['email_cc'], ['test.😊.cc@example.com'])
         self.assertEqual(self._mails[0]['email_to'], ['test.😊@example.com'])
 
+    @mute_logger('odoo.addons.mail.models.mail_mail')
+    def test_mail_mail_send_server_attachment_to_download_link(self):
+        """ Test that when the mail size exceeds the max email size limit,
+        attachments are turned into download links added at the end of the
+        email content.
+
+        The feature is tested in the following conditions:
+        - using a specified server or the default one (to test command ICP parameter)
+        - in batch mode
+        - with mail that exceed (with one or more attachments) or not the limit
+        - with attachment owned by a business record or not: attachments not owned by a
+        business record are never turned into links because their lifespans are not
+        controlled by the user (might even be deleted right after the message is sent).
+        """
+        def count_attachments(message):
+            return (0 if isinstance(message, str)
+                    else sum(count_attachments(part) for part in message.get_payload()) if message.is_multipart()
+                    else 'attachment' in message.get('Content-Disposition', ''))
+
+        attachment_size_in_bytes = 1024 * 128
+        datas = base64.b64encode((''.join('.' for _ in range(0, attachment_size_in_bytes))).encode())
+        # Define some constant to ease the understanding of the test
+        test_mail_server = self.mail_server_domain_2
+        cst_max_size_always_exceed = 0.1
+        cst_max_size_never_exceed = 10
+
+        # cst_max... are used to define max_email_size on the server that is not used with value that would
+        # lead to an incorrect expected_is_links if they were used. We do this to check that the right max
+        # email size is used.
+        for (n_attachment, mail_server, default_email_size, max_email_size, business_attachment), expected_is_links in (
```

Comment:
Note entirely sure why "expected_is_links" is not in the initial tuple, I'm guessing to split data/assertion, but not sure it's worth as it's adding a lot of parenthesis everywhere ^^